### PR TITLE
github: Also run ci-tests workflow on macOS

### PR DESCRIPTION
**Pull request by peterjgilbert**
- [x] ci: run tests in macOS
- [ ] add darwin build to goreleaser config

### CODE REVIEW

**Comment by peterjgilbert** [comment]
Tested both the build and release locally with current oasis-core master. Should be good to go after we cut a new release of oasis-core and update the oasis-core dependency.

File: .goreleaser.yml [hunk 1]
```diff
@@ -25,6 +25,7 @@ builds:
       - "{{.Env.GOLDFLAGS_VERSION}}"
     goos:
       - linux
+      - darwin
```
Comment (kostko) [inline .goreleaser.yml:28]: Can we confirm just adding this here works?

Comment (peterjgilbert) [reply .goreleaser.yml:28]: It worked locally for me on Linux with the same version of goreleaser we use in CI (0.141.0).

Comment (peterjgilbert) [reply .goreleaser.yml:28]:
```
# ls dist/
SHA256SUMS-1.0.0-dev.txt    oasis-core-ledger_darwin_amd64
config.yaml                 oasis-core-ledger_linux_amd64
ledger-signer_darwin_amd64  oasis_core_ledger_1.0.0-dev_darwin_amd64.tar.gz
ledger-signer_linux_amd64   oasis_core_ledger_1.0.0-dev_linux_amd64.tar.gz
```

Comment (peterjgilbert) [reply .goreleaser.yml:28]: ^ this was using the `oasisprotocol/oasis-core-dev:master` docker image

Comment (tjanez) [reply .goreleaser.yml:28]: Cool. I would also like to test the GitHub Actions workflow via my private fork.

Comment (kostko) [reply .goreleaser.yml:28]: Yeah I meant the GitHub Actions workflow :)

**Review by tjanez — APPROVED** [review]
Looks good!

**Pull request merged**